bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,351 @@
+                nodeIndex, userIndex, nodes);
+    }
+
+    /*
+     * public String getGatewayIP() {
+     * bridge = (Bridge) thingGeneric.getBridgeUID();
+     * if (thingGeneric.getBridgeUID() != null) {
+     * return ((SoulissGatewayHandler) (Bridge) this.getThingByUID(bridge.getUID()).getHandler()).ipAddressOnLAN;
+     * }
+     * return null;
+     * }
+     */
+
+    public void setNodes(int nodes) {
+        this.nodes = nodes;
+    }
+
+    int iPosNodeSlot = 2;
+
+    public int getNodes() {
+        Thing thing;
+        int maxNode = 0;
+        Iterator<Thing> iterator = bridge.getThings().iterator();
+        while (iterator.hasNext()) {
+            thing = iterator.next();
+            String[] uuidStrings = thing.getUID().getAsString().split(SoulissBindingConstants.UUID_NODE_SLOT_SEPARATOR);
+            String[] uuidNodeNumber = uuidStrings[0].split(SoulissBindingConstants.UUID_ELEMENTS_SEPARATOR);
+
+            iPosNodeSlot = 2; // if uuid is of type souliss:gateway:[typical]:[node]-[slot] then node/slot is at
+                              // position 2
+            if (uuidNodeNumber.length > 3) {
+                iPosNodeSlot = 3;
+            }
+            if (Integer.parseInt(uuidNodeNumber[iPosNodeSlot]) > maxNode) {
+                maxNode = Integer.parseInt(uuidNodeNumber[iPosNodeSlot]);
+            }
+            // alla fine la lunghezza della lista sarà uguale al numero di nodi presenti
+        }
+        return maxNode + 1;
+    }
+
+    // public void setMaxnodes(int maxnodes) {
+    // this.maxnodes = maxnodes;
+    // }
+
+    public void setMaxTypicalXnode(int maxTypicalXnode) {
+        this.maxTypicalXnode = maxTypicalXnode;
+    }
+
+    // public void setMaxrequests(int maxrequests) {
+    // this.maxrequests = maxrequests;
+    // }
+
+    public int getMaxTypicalXnode() {
+        return maxTypicalXnode;
+    }
+
+    /**
+     * The {@link gatewayDetected} is used to notify that UDPServer decoded a Ping Response from gateway
+     *
+     * @author Tonino Fazio - Initial contribution
+     * @author Luca Calcaterra - Refactor for OH3
+     */
+
+    public void gatewayDetected() {
+        logger.debug("Setting Gateway ONLINE");
+        try {
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            logger.debug("Illegal status transition to ONLINE");
+        }
+        countPingKo = 0; // reset counter
+    }
+
+    public void pingSent() {
+        if (++countPingKo > 3) {
+            // if GW do not respond to ping it is setted to OFFLINE
+            logger.debug("Gateway do not respond to {} ping packet - setting OFFLINE", countPingKo);
+            if (bridge.getHandler() != null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Gateway "
+                        + bridge.getHandler().getThing().getUID() + " do not respond to " + countPingKo + " ping");
+            }
```

Comment:
Please read this
https://www.openhab.org/docs/developer/guidelines.html#f-logging
no need to log it twice as changing the thing status will do event.log